psydac/linalg/topetsc.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+
+from psydac.linalg.block import BlockVectorSpace, BlockVector, BlockMatrix
+from psydac.linalg.stencil import StencilVectorSpace, StencilVector
+from scipy.sparse import coo_matrix, bmat
+
+from mpi4py import MPI
+
+def flatten_vec( vec ):
+    """ Return the flattened 1D array values and indices corresponding to the given vector.
```

Comment:
I don't fully understand this description.

When in Psydac we convert a `StencilVector` to a `numpy.ndarray`, the result is always a dense vector with the size of the global (non distributed) array. As a consequence, in the parallel case the result is an array with many zeros and few non-zero entries.

Does `flatten_vec` return only the non-zero entries instead? If so, this should be clarified in the docstring.

Reply:
it returns all the entries owned by the process even the zero one,  so it's not like `StencilVector`

Reply:
I changed it to this `Return the flattened 1D array values and indices owned by the process of the given vector.`